File: 169. Majority Element.java
class Solution {
    public int majorityElement(int[] nums) {
        // sort list的语法
        Arrays.sort(nums);
        // 1.获得list长度，2.整除用一个斜杠，3.使用index获得元素
        return nums[nums.length/2];
    }
}

File: 169. Majority Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums = sorted(nums)
        return nums[len(nums)//2]


# boyer-moore voting algorithm
# time n space 1
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 0
        candidate = None

        for num in nums:
            if count == 0:
                candidate = num
            count += (1 if num == candidate else -1)

        return candidate